File: backend/coffee-inventory-api/Data/InventoryItemDTO.cs
namespace API.Data;

public class InventoryItemDTO
{
    public string Id { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Roaster { get; set; } = string.Empty;
    public string Origin { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    public decimal AmountUsed { get; set; }

    public decimal Price { get; set; }

    public InventoryItemTypeDTO Type { get; set; }

    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
}

public enum InventoryItemTypeDTO
{
    Coffee = 0,
    Other = 1
}

File: frontend/src/services/entities/inventoryItemDTO.ts
export interface InventoryItemDTO {
    id: string,

    name: string,

    roaster: string,
    origin: string,

    type: InventoryItemTypeDTO,
    startDate: Date,
    endDate?: Date,

    amount: number,
    amountUsed: number,

    price: number
}

export enum InventoryItemTypeDTO {
    Coffee = 0,
    Other = 1
}
